src/lib.rs

```diff
@@ -11,6 +11,898 @@
+                            ANY_MAJOR,
+                            vsomeip::ANY_MINOR,
+                        );
+                        request_single_event_safe(
+                            application_wrapper,
+                            service_id,
+                            instance_id,
+                            event_id,
+                            event_id,
+                        );
+                        get_pinned_application(application_wrapper).subscribe(
+                            service_id,
+                            instance_id,
+                            event_id,
+                            ANY_MAJOR,
+                            event_id,
+                        );
+                    }
+                }
+                register_message_handler_fn_ptr_safe(
+                    application_wrapper,
+                    service_id,
+                    instance_id,
+                    event_id,
+                    msg_handler,
+                );
+
+                // Letting vsomeip settle
+                tokio::time::sleep(Duration::from_millis(5)).await;
+
+                trace!(
+                    "{}:{} - Registered vsomeip message handler.",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                );
+
+                Ok(())
+            }
+            RegistrationType::Request(_) => {
+                trace!(
+                    "{}:{} - Registering for Request style messages.",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                );
+                let Some(sink_filter) = sink_filter else {
+                    return Err(UStatus::fail_with_code(
+                        UCode::INVALID_ARGUMENT,
+                        "Unable to map source and sink filters to uProtocol message type",
+                    ));
+                };
+
+                let (_, service_id) = split_u32_to_u16(sink_filter.ue_id);
+                let instance_id = 1; // TODO: Set this to 1? To ANY_INSTANCE?
+                let (_, method_id) = split_u32_to_u16(sink_filter.resource_id);
+
+                trace!(
+                    "{}:{} - register_message_handler: service: {} instance: {} method: {}",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                    service_id,
+                    instance_id,
+                    method_id
+                );
+
+                {
+                    let mut offered_services = OFFERED_SERVICES.write().await;
+                    if !offered_services.contains(&(service_id, instance_id, method_id)) {
+                        get_pinned_application(application_wrapper).offer_service(
+                            service_id,
+                            instance_id,
+                            ANY_MAJOR,
```

Comment:
You must specify the major / minor version when offering the service. They are optional in vsomeip, but default values are [(0, 0)](https://github.com/COVESA/vsomeip/blob/master/interface/vsomeip/application.hpp#L171)